File: ZAB_FYP_PORTAL FLUTTER FRONTEND/lib/models/advisor_model.dart
import 'area_model.dart';

class AdvisorModel {
  int? iD;
  String? fIRSTNAME;
  String? lASTNAME;
  String? eMAIL;
  String? pROFILEPIC;
  String? dESIGNATION;
  String? roomNo;
  String? campus;
  List<AreaModel>? areas;

  AdvisorModel(
      {this.iD,
      this.fIRSTNAME,
      this.lASTNAME,
      this.eMAIL,
      this.pROFILEPIC,
      this.dESIGNATION,
      this.roomNo,
      this.campus,
      this.areas});

  AdvisorModel.fromJson(Map<String, dynamic> json) {
    iD = json['ID'];
    fIRSTNAME = json['FIRST_NAME'];
    lASTNAME = json['LAST_NAME'];
    eMAIL = json['EMAIL'];
    dESIGNATION = json['DESIGNATION'];
    pROFILEPIC = json['PROFILE_PIC'];
    roomNo = json['RoomNo'];
    campus = json['Campus'];
    if (json['Areas'] != null) {
      areas = <AreaModel>[];
      json['Areas'].forEach((v) {
        areas!.add(new AreaModel.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['ID'] = this.iD;
    data['FIRST_NAME'] = this.fIRSTNAME;
    data['LAST_NAME'] = this.lASTNAME;
    data['EMAIL'] = this.eMAIL;
    data['DESIGNATION'] = this.dESIGNATION;
    data['PROFILE_PIC'] = this.pROFILEPIC;
    data['RoomNo'] = this.roomNo;
    data['Campus'] = this.campus;
    if (this.areas != null) {
      data['Areas'] = this.areas!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

File: ZAB FYP NODE JS BACKEND/src/models/advisor.model.ts
import User from "./user.model";
import Area from "./area.model";


export default class Advisor extends User {
  public DESIGNATION?: string ;
  public RoomNo?: string ;
  public Campus?: string ;
  public Areas?: Array<Area> ;
  constructor(ID: string, USERNAME: string, PASSWORD: string, FIRST_NAME: string, LAST_NAME: string,
    EMAIL: string, PHONE_NUMBER: string, STATUS: string, ROLE: string, PROFILE_PIC: string, IS_VERIFIED: boolean,
    IS_LOCKED: boolean, EXTRA_PROPERTIES: string,
    INSTITUTEID: string, AREAS: Array<Area> | null=null)  {

    super(ID,USERNAME,PASSWORD,FIRST_NAME,LAST_NAME,EMAIL,PHONE_NUMBER,STATUS,ROLE,IS_VERIFIED,IS_LOCKED,PROFILE_PIC,undefined,INSTITUTEID);

    if(EXTRA_PROPERTIES)
    {
      var prop=JSON.parse(EXTRA_PROPERTIES);
      this.DESIGNATION=prop.Designation;
      this.RoomNo=prop.RoomNo;
      this.Campus=prop.Campus;
    }
    
    if(AREAS)
    this.Areas=AREAS;

    
  }

}
